tests/test_buffer_protocol.rs

```diff
@@ -132,3 +133,93 @@ fn test_buffer_referenced() {
 
     assert!(drop_called.load(Ordering::Relaxed));
 }
+
+#[pyclass]
+struct ReleaseBufferError {}
+
+static BUF_BYTES: &[u8] = b"hello world";
+
+#[pymethods]
+impl ReleaseBufferError {
+    unsafe fn __getbuffer__(
+        slf: PyRef<'_, Self>,
+        view: *mut ffi::Py_buffer,
+        flags: c_int,
+    ) -> PyResult<()> {
+        if view.is_null() {
+            return Err(PyBufferError::new_err("View is null"));
+        }
+
+        if (flags & ffi::PyBUF_WRITABLE) == ffi::PyBUF_WRITABLE {
+            return Err(PyBufferError::new_err("Object is not writable"));
+        }
+
+        (*view).obj = ffi::_Py_NewRef(slf.as_ptr());
+
+        (*view).buf = BUF_BYTES.as_ptr() as *mut c_void;
+        (*view).len = BUF_BYTES.len() as isize;
+        (*view).readonly = 1;
+        (*view).itemsize = 1;
+
+        (*view).format = if (flags & ffi::PyBUF_FORMAT) == ffi::PyBUF_FORMAT {
+            let msg = CString::new("B").unwrap();
+            msg.into_raw()
+        } else {
+            ptr::null_mut()
+        };
+
+        (*view).ndim = 1;
+        (*view).shape = if (flags & ffi::PyBUF_ND) == ffi::PyBUF_ND {
+            &mut (*view).len
+        } else {
+            ptr::null_mut()
+        };
+
+        (*view).strides = if (flags & ffi::PyBUF_STRIDES) == ffi::PyBUF_STRIDES {
+            &mut (*view).itemsize
+        } else {
+            ptr::null_mut()
+        };
+
+        (*view).suboffsets = ptr::null_mut();
+        (*view).internal = ptr::null_mut();
+
+        Ok(())
+    }
+
+    unsafe fn __releasebuffer__(&self, _view: *mut ffi::Py_buffer) -> PyResult<()> {
+        Err(PyValueError::new_err("oh dear"))
+    }
+}
+
+#[test]
+fn test_releasebuffer_unraisable_error() {
+    static UNRAISABLE_CALLED: Mutex<Option<(PyErr, PyObject)>> = const_mutex(None);
```

Comment:
This is really a nit as the effort might not be worth it at all, but wouldn't it be possible to use a `#[pyclass]` to capture the unraisable value without resorting to global state by setting the unraisablehook to use a method call? (I am not sure if we have the API to bind the self poarameter though?)